src/components/tl/nccl/allgatherv/allgatherv.c

```diff
@@ -93,6 +93,244 @@ ucc_status_t ucc_tl_nccl_allgatherv_p2p_start(ucc_coll_task_t *coll_task)
     return status;
 }
 
+typedef struct {
+    void *cpu_sbuf;
+    void *staged_sbuf;
+    uintptr_t sbuf_len;
+
+    int first_peer_rank;
+    void *first_peer_cpu_rbuf;
+    uintptr_t first_peer_len;
+
+    int last_peer_rank;
```

Comment:
int -> ucc_rank_t 
(during initialization set to UCC_RANK_INVALID instead of -1)